pylib/cqlshlib/copyutil.py

```diff
@@ -2101,6 +2101,13 @@ def convert_map(val, ct=cql_type):
             return ImmutableDict(frozenset((convert_mandatory(ct.subtypes[0], v[0]), convert(ct.subtypes[1], v[1]))
                                  for v in [split(split_format_str % vv, sep=sep) for vv in split(val)]))
 
+        def convert_vector(val, ct=cql_type):
+            string_coordinates = val.strip("[]").split(',')
```

Comment:
Semi-nit: this may not be quite as precise as we'd like.  strip() will remove any leading or trailing characters matching the arguments.  Upshot is that this:

```
"[1,2,3,4,5,6].strip("[]")
```

is equivalent to this:

```
"[[1,2,3,4,5,6]]".strip("[]")
```

which may not be what we want, although to some degree that depends on how much we care about malformed input (and what we should do if/when we discover it).

It looks like the other functions in ImportConversion are largely using [split()](https://github.com/datastax/cassandra/blob/ds-trunk/pylib/cqlshlib/copyutil.py#L1987) to do this kind of thing.  Might be useful to consider that here.  Another nice side effect is that split() appears to traverse the list only once while strip().split() will do so twice.

Reply:
I think using [split()](https://github.com/datastax/cassandra/blob/ds-trunk/pylib/cqlshlib/copyutil.py#L1987) function is a better approach than relying on the combination I have proposed initially, it is widely used in other functions already and as you pointed out it does the job in a single collection interation.